### ISSUE TITLE
Feature Request: Don't Sort lists with elements that do not implement IComparable<T>

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-dotnet#2395]
### Description
Sorting elements in a list should be meaningful. The method `List<T>.Sort()` should therefor only be called if `T` implements `IComparable<T>`. Obviously, this rule should not apply to `List<T>.Sort(IComparer<T>)`, as the comparer handles the sorting. This should both be available in C# and VB.

### Expected behavior
``` C#
using System;
using System.Collections.Generic;

namespace SonarAnalyzer.Experiments.CSharp
{
    public class DontSortUncomparebleLists
    {
        public DontSortUncomparebleLists()
        {
            var objects = new List<object>();
            objects.Sort(); // noncompliant. object does not implement IComparable<T>
            objects.Sort(Comparer<object>.Default); // compliant. The comparer should no what to do.

            var ints = new List<int>();
            ints.Sort(); // compliant. int implements IComparable<int>
        }

        public static void Sort<T>(List<T> list)
        {
            list.Sort(); // noncompliant. object might not implement IComarable<T>
        }

        public static void Sort2<T>(List<T> list)
            where T : IComparable<T>
        {
            list.Sort(); // compliant. T implements IComparable<T>
        }
    }
}
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1992](https://sonarsource.atlassian.net/browse/NET-1992)

[NET-1992]: https://sonarsource.atlassian.net/browse/NET-1992?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
